### PR TITLE
Allow markdown extensions (autolink)

### LINKED ISSUE #108: [markdown] add support for extensions
There are a few nice extensions:
https://github.com/commonmark/commonmark-java?tab=readme-ov-file#extensions

Let's create a spi to provide an extension to the parser and then create new artifacts for each relevant extensions (and maybe a grouped one)

### PR DESCRIPTION
**Pull request by mcruzdev**
Fixes #108

### CODE REVIEW

**Comment by mcruzdev** [comment]
Hi 👋

Could you review this PR?

cc: @mkouba @ia3andy

File: markdown/autolink/deployment/src/main/java/io/quarkiverse/qute/web/markdown/autolink/deployment/QuteWebMarkdownAutolinkProcessor.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.FeatureBuildItem;
+
+class QuteWebMarkdownAutolinkProcessor {
+
+    private static final String FEATURE = "qute-web-markdown-autolink";
+
+    @BuildStep
+    FeatureBuildItem feature() {
+        return new FeatureBuildItem(FEATURE);
```
Comment (ia3andy) [inline markdown/autolink/deployment/src/main/java/io/quarkiverse/qute/web/markdown/autolink/deployment/QuteWebMarkdownAutolinkProcessor.java:15]: Let's not add it as a feature, it's very small and the feature is markdown

**Review by ia3andy — APPROVED** [review]
LGTM (just one minor change)

**Review by mkouba — COMMENTED** [review]
Shouldn't we mention this in the docs?

**Review by ia3andy — APPROVED** [review]
LGTM

**Pull request merged**